lib/vanagon/platform/windows.rb

```diff
@@ -119,8 +119,9 @@ def generate_msi_package(project)
         #   -gg             - Generate GUIDS now
         #   -dr             - Directory reference to root directories (cannot contains spaces e.g. -dr MyAppDirRef)
         #   -sreg           - Suppress registry harvesting.
-        #   -var <variable> - Substitute File/@Source="SourceDir" with a preprocessor or a wix variable
-        "cd $(tempdir); \"$$WIX/bin/heat.exe\" dir staging -v -ke -indent 2 -cg #{cg_name} -gg -dr #{dir_ref} -t wix/#{project.name}.filter.xslt -sreg -var var.StageDir -out wix/#{project.name}-harvest.wxs",
+        "cd $(tempdir); \"$$WIX/bin/heat.exe\" dir staging -v -ke -indent 2 -cg #{cg_name} -gg -dr #{dir_ref} -t wix/#{project.name}.filter.xslt -sreg -out wix/#{project.name}-harvest.wxs",
+        # candle individual files to generate the wxo files.
+        "cd $(tempdir); \"$$WIX/bin/candle.exe\"  -arch #{@architecture} -out wix/#{project.name}-harvest.wixobj   wix/#{project.name}-harvest.wxs  ",
```

Comment:
Do we need `-dPlatform=#{@architecture}` or any of the version information passed in with https://github.com/puppetlabs/puppet_for_the_win/blob/master/tasks/windows/windows.rake#L23 ? What do `-ext WiXUtilExtension -ext WixUIExtension` do any why don't we need them anymore?

It also looks like candle is called on all the wxs files (https://github.com/puppetlabs/puppet_for_the_win/blob/master/tasks/windows/windows.rake#L271-L285). Are we also going to have to do that? Should we glob for all the files in the `wxs` directory and run candle on all of them?